### PR TITLE
Use f-strings where possible

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
It's faster.

Also a couple small changes here and there.

### CODE REVIEW

File: nmrglue/analysis/linesh.py [hunk 1]
```diff
@@ -496,8 +496,7 @@ def fit_NDregion(region, lineshapes, params, amps, bounds=None,
     if wmask is None:   # default is to include all points in region
         wmask = np.ones(shape, dtype='bool')
     if wmask.shape != shape:
-        err = "wmask has incorrect shape:" + str(wmask.shape) +   \
-              " should be " + str(shape)
+        err = f"wmask has incorrect shape: {wmask.shape} should be {shape}"
```
Comment (DimitriPapadopoulos) [inline nmrglue/analysis/linesh.py:499]: I have added a space after `incorrect shape:`

File: nmrglue/fileio/bruker.py [hunk 2]
```diff
@@ -1546,7 +1546,7 @@ def read_binary(filename, shape=(1), cplex=True, big=True, isfloat=False):
         return dic, data.reshape(shape)
 
     except ValueError:
-        warn(str(data.shape) + "cannot be shaped into" + str(shape))
+        warn(f"{data.shape} cannot be shaped into {shape}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/bruker.py:1549]: I have added a space before and after "cannot be shaped into".

File: nmrglue/fileio/bruker.py [hunk 3]
```diff
@@ -2169,9 +2169,9 @@ def read_jcamp(filename, encoding=locale.getpreferredencoding()):
                     key, value = parse_jcamp_line(line, f)
                     dic[key] = value
                 except:
-                    warn("Unable to correctly parse line:" + line)
+                    warn(f"Unable to correctly parse line: {line}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/bruker.py:2172]: I have added a space after `line:`.

File: nmrglue/fileio/bruker.py [hunk 4]
```diff
@@ -2169,9 +2169,9 @@ def read_jcamp(filename, encoding=locale.getpreferredencoding()):
...
             else:
-                warn("Extraneous line:" + line)
+                warn(f"Extraneous line: {line}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/bruker.py:2174]: I have added a space after `line:`.

File: nmrglue/fileio/jcampdx.py [hunk 5]
```diff
@@ -85,7 +85,7 @@ def _parsejcampdx(filename):
                 key = _getkey(currentkey)
                 value = "".join(currentvaluestrings)  # collapse
                 if not value.strip():
-                    warn("JCAMP-DX key without value:" + key)
+                    warn(f"JCAMP-DX key without value: {key}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/jcampdx.py:88]: I have added a space after `value:`.

File: nmrglue/fileio/jcampdx.py [hunk 6]
```diff
@@ -121,7 +121,7 @@ def _parsejcampdx(filename):
                 keystr = keysplit[0][2:]  # remove "##" already here
                 valuestr = keysplit[1]
                 if not keystr:
-                    warn("Empty key in JCAMP-DX line:" + line)
+                    warn(f"Empty key in JCAMP-DX line: {line}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/jcampdx.py:124]: I have added a space after `line:`.

File: nmrglue/fileio/jcampdx.py [hunk 7]
```diff
@@ -134,7 +134,7 @@ def _parsejcampdx(filename):
         else:
             if activeblock:
                 if currentkey is None:
-                    warn("JCAMP-DX data line without associated key:" + line)
+                    warn(f"JCAMP-DX data line without associated key: {line}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/jcampdx.py:137]: I have added a space after `line:`.

File: nmrglue/fileio/jcampdx.py [hunk 8]
```diff
@@ -272,7 +272,7 @@ def _parse_affn_pac(datalines):
             try:
                 value = float(base + (exp if exp is not None else ""))
             except ValueError:
-                warn("Data parsing failed at line:" + dataline)
+                warn(f"Data parsing failed at line: {dataline}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/jcampdx.py:275]: I have added a space after `line:`.

File: nmrglue/fileio/jcampdx.py [hunk 9]
```diff
@@ -399,7 +398,7 @@ def _parse_pseudo(datalines):
                                                          value_to_append,
                                                          data)
                 if not success:
-                    warn("Data parsing failed at line:" + dataline)
+                    warn(f"Data parsing failed at line: {dataline}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/jcampdx.py:401]: I have added a space after `line:`.

File: nmrglue/fileio/varian.py [hunk 10]
```diff
@@ -504,8 +504,8 @@ def find_torder(dic, shape):
             warn("missing phase order, torder set to 'r'")
             return 'r'
 
-    warn("No trace ordering for" + str(ndim) +
-         "dimensional data, torder set to 'r'")
+    warn(f"No trace ordering for {ndim}dimensional data, "
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/varian.py:507]: I have added a space after `ordering for`.

File: nmrglue/fileio/varian.py [hunk 11]
```diff
@@ -523,7 +523,7 @@ def torder2i2t(torder):
     elif torder in ('regular', 'r'):
         return fileiobase.index2trace_reg
     else:
-        raise ValueError("unknown torder" + str(torder))
+        raise ValueError(f"unknown torder {torder}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/varian.py:526]: I have added a space after `torder`.

File: nmrglue/fileio/varian.py [hunk 12]
```diff
@@ -539,7 +539,7 @@ def torder2t2i(torder):
     elif torder in ('regular', 'r'):
         return fileiobase.trace2index_reg
     else:
-        raise ValueError("unknown torder" + str(torder))
+        raise ValueError(f"unknown torder {torder}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/varian.py:542]: I have added a space after `torder`.

File: nmrglue/fileio/varian.py [hunk 13]
```diff
@@ -571,7 +571,7 @@ def reorder_data(data, shape, torder):
         try:
             data = data.reshape(shape)
         except ValueError:
-            warn(str(data.shape) + "cannot be shaped into" + str(shape))
+            warn(f"{data.shape} cannot be shaped into {shape}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/varian.py:574]: I have added a space before and after `cannot be shaped into`.

File: nmrglue/fileio/varian.py [hunk 14]
```diff
@@ -716,14 +716,14 @@ def read_fid(filename, shape=None, torder='flat', as_2d=False,
         try:
             return dic, reorder_data(data, shape, torder)
         except:
-            warn("data cannot be re-ordered, returning raw 2D data\n" +
-                 "Provided shape: " + str(shape) + " torder: " + str(torder))
+            warn(f"data cannot be re-ordered, returning raw 2D data\n" +
+                 f"Provided shape: {shape} torder: {torder}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/varian.py:720]: I have added a space before `torder:`.

File: nmrglue/fileio/varian.py [hunk 15]
```diff
@@ -716,14 +716,14 @@ def read_fid(filename, shape=None, torder='flat', as_2d=False,
...
             return dic, data
 
     try:
         data = data.reshape(shape)
     except ValueError:
-        warn(str(data.shape) + "cannot be shaped into" + str(shape))
+        warn(f"{data.shape} cannot be shaped into {shape}")
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/varian.py:726]: I have added a space before and after `cannot be shaped into`.

File: nmrglue/fileio/varian.py [hunk 16]
```diff
@@ -858,7 +858,7 @@ def read_fid_ntraces(filename, shape=None, torder='flat', as_2d=False,
...
```
Comment (DimitriPapadopoulos) [inline nmrglue/fileio/varian.py:861]: I have added a space before and after `cannot be shaped into`.

File: nmrglue/util/xcpy.py [hunk 17]
```diff
@@ -273,7 +273,7 @@ def run(cpython, script, pass_current_folder=True, use_shell=None, dry=None):
     args = [cpython, script] + cd
 
     if dry:
-        MSG("The following command will be executed: \n" + " ".join(args))
+        MSG("The following command will be executed:\n" + " ".join(args))
```
Comment (DimitriPapadopoulos) [inline nmrglue/util/xcpy.py:276]: I have removed the spurious space before `\n`. I am wondering whether the intent was to have `\n` after `" ".join(args)`.

Comment (jjhelmus) [reply nmrglue/util/xcpy.py:276]: I'm not sure on the intent here.  Likely just a typo or overlooked extra character.

**Comment by jjhelmus** [comment]
Thanks @DimitriPapadopoulos for the PR.  I like how f-string make the message more understandable.

**Pull request merged**